### PR TITLE
Set correct module for APIs in torch module

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #75801

### CODE REVIEW

File: torch/__init__.py [hunk 1]
```diff
@@ -228,10 +228,12 @@ def _load_global_deps():
...
 
+for name in dir(_C):
+    if name[0] != '_' and not name.endswith('Base'):
+        obj = getattr('torch._C', name)
+        obj.__module__ = 'torch'
 
-__all__ += [name for name in dir(_C)
-            if name[0] != '_' and
-            not name.endswith('Base')]
+__all__ += [name ]
```
Comment (albanD) [inline torch/__init__.py:236]: missing indentation

File: torch/__init__.py [hunk 2]
```diff
@@ -742,6 +744,8 @@ def manager_path():
 for name in dir(_C._VariableFunctions):
     if name.startswith('__') or name in PRIVATE_OPS:
         continue
+    obj = getattr(_C._VariableFunctions, name)
```
Comment (albanD) [inline torch/__init__.py:747]: nit: re-use this obj below to avoid re-doing the getattr

File: torch/__init__.py [hunk 3]
```diff
@@ -228,10 +229,14 @@ def _load_global_deps():
...
-            if name[0] != '_' and
-            not name.endswith('Base')]
+for name in dir(_C):
+    if name[0] != '_' and not name.endswith('Base'):
+        __all__ += name
+        obj = getattr(_C, name)
+        if (isinstance(obj, Callable) or inspect.isclass(obj)) and (obj.__module__ != 'torch'):
+            # TODO: fix their module from C++ side
+            if name not in ['DisableTorchFunction', 'Generator']:
+                setattr(obj, '__module__', 'torch')
```
Comment (albanD) [inline torch/__init__.py:239]: lint comment sounds good, we can do `obj.__module__ = 'torch'` here.

**Review by albanD — APPROVED** [review]
lint needs fixing.
SGTM otherwise.

File: test/test_public_bindings.py [hunk 4]
```diff
@@ -317,7 +317,7 @@ def looks_public_or_not(elem, modname, mod, is_public=True):
                 if not (isinstance(obj, Callable) or inspect.isclass(obj)):
                     return
                 elem_module = getattr(obj, '__module__', None)
-                elem_modname_starts_with_mod = elem_module is not None and elem_module.startswith(modname)
+                elem_modname_starts_with_mod = elem_module is not None and elem_module.startswith(modname) and '._' not in elem_module
```
Comment (anjali411) [inline test/test_public_bindings.py:320]: TODO: update the wiki

**Review by albanD — COMMENTED** [review]
Thanks!

**Pull request closed without merging**